### PR TITLE
Bug fix for BasketCalculate when username or is_anonymous is not in url

### PR DESCRIPTION
**Pull request by BenjiLee**
https://openedx.atlassian.net/browse/LEARNER-5285

### CODE REVIEW

File: ecommerce/extensions/api/v2/tests/views/test_baskets.py [hunk 1]
```diff
@@ -567,39 +566,14 @@ def test_basket_calculate_anonymous_skip_lms(self, mock_get_lms_resource_for_use
...
     def test_basket_calculate_anonymous_caching(self, mock_calculate_basket):
         """Verify a request made with the is_anonymous parameter is cached"""
-        url_with_one_sku = self._generate_sku_url(self.products, number_of_products=1, username=None)
-        url_with_two_skus = self._generate_sku_url(self.products, number_of_products=2, username=None)
+        url_with_one_sku = self._generate_sku_url(self.products[0:1], username=None)
+        url_with_two_skus = self._generate_sku_url(self.products[0:2], username=None)
         url_with_two_skus_reversed = self._generate_sku_url(
-            self.products, number_of_products=2, username=None, reverse_skus=True
-        )
+            [self.products[0], self.products[1]], username=None)
```
Comment (robrap) [inline ecommerce/extensions/api/v2/tests/views/test_baskets.py]: Nit: Does this fit on one line or could the ending parentheses remain on its own line?

File: ecommerce/extensions/api/v2/tests/views/test_baskets.py [hunk 2]
```diff
@@ -627,38 +599,59 @@ def test_basket_calculate_anonymous_caching(self, mock_calculate_basket):
...
         response = self.client.get(url_with_one_sku_no_anon)
         self.assertEqual(response.status_code, 200)
         self.assertFalse(mock_calculate_basket.called, msg='The cache should be hit.')
         self.assertEqual(response.data, expected)
 
+    @mock.patch('ecommerce.extensions.api.v2.views.baskets.logger.warning')
+    @mock.patch('ecommerce.extensions.api.v2.views.baskets.BasketCalculateView._calculate_temporary_basket')
+    @override_flag('disable_calculate_temporary_basket_atomic_transaction', active=True)
+    def test_basket_calculate_no_query_parameters(self, mock_calculate_basket, mock_logger):
+        """Verify a request made without query parameters uses the request user"""
```
Comment (robrap) [inline ecommerce/extensions/api/v2/tests/views/test_baskets.py:634]: Should we add documentation here that this is a case that we don't want to support in any newer version of the endpoint?

**Review by robrap — APPROVED** [review]
See comments.

File: ecommerce/extensions/api/v2/tests/views/test_baskets.py (hunk 1 above)
Comment (BenjiLee) [reply ecommerce/extensions/api/v2/tests/views/test_baskets.py]: will fix

File: ecommerce/extensions/api/v2/tests/views/test_baskets.py (hunk 2 above)
Comment (BenjiLee) [reply ecommerce/extensions/api/v2/tests/views/test_baskets.py:634]: I'd rather this test fail when that time comes. Additionally, I don't know when or if that time will come. I'd prefer to leave it out.

**Pull request merged**